### ISSUE TITLE
a11y: Fix keyboard navigation, label and focus display issues on the sortable table

### ISSUE THREAD

**Issue by nwmac** [rancher/dashboard#12794]
a11y Issues: 61, 87

**Issue 61 - Sort functionality not programmatically identified**

Should be fixed in general on the sortable table. Affects many uses of the table - example below:

![image](https://github.com/user-attachments/assets/314d7ca4-2f68-4e89-94ce-0005da780035)

When using a screen reader, users should understand the current sorting state applied to the data within the table.

Ensure that the sort state is programmatically defined for the sortable columns.
Apply the following changes:
• Provide the sort state using the "aria-sort" attribute on the <th> element. 
o Set the value of "aria-sort" to "ascending" when the content is sorted in ascending order. 
o Set the value to "descending" when the content is sorted in descending order. 
o If no sorting is applied on the content, set the value to "none". 
o Ensure that the sort state is updated on user interaction. 

For information on accessible sort controls, visit https://www.w3.org/WAI/ARIA/apg/patterns/table/examples/sortable-table/

**Issue 87 - The checkboxes do not have the label associated with them**

![image](https://github.com/user-attachments/assets/d86379a0-c49c-4ff7-9418-56f194ad10ad)

Example uses:

Account and API Keys
 - API keys section
  - Checkboxes in table

Projects/Namespaces
- Checkboxes in table

Apply the following changes:
• Specify the "id" attribute with a unique value to <td> element containing "kubeconfig-user-2rgrpmfj4q", "kubeconfig-user-2rgrpp4hn6", kubeconfig-user-2rgrptgkvb", "kubeconfig-user-2rgrpw54zp" and so on text.
• Specify "aria-labelledby" attribute to <span> element containing checkboxes.
• Reference the "id" attribute values (of <td> element) via "aria-labelledby" attribute.

**Comment by izaac** [comment]
@aalves08 is this requirement in the issue description going to be implemented too?

> Provide the sort state using the "aria-sort" attribute on the element.

The Table headers are picked by the reader but the sort state isn't at the current implementation state.

**Comment by izaac** [comment]
There are a few readers out there but the easiest one to get working is this chrome extension which doesn't work on Brave for some reason.  https://chromewebstore.google.com/detail/screen-reader/kgejglhpjiefppelpmljglcjbhoiplfn

**Comment by aalves08** [comment]
@izaac I did a bit of research on this and for Mac OS the most popular screen reader seems to be it's own app called `VoiceOver` (I got mac OS system). For windows you have `JAWS` and `NVDA`. Since I can only test in mac OS, here's the output of the screen reader when tested against the ARIA page for sortable tables https://www.w3.org/WAI/ARIA/apg/patterns/table/examples/sortable-table/:

https://github.com/user-attachments/assets/6f9eecb2-0a9c-410c-a1f4-e3f7a1145b1e

Not even on this page the screen reader picks up the sorting, which ARIA implementations are flawless and, as you can check, `aria-sort` is implemented the same way as we have on our code.

I don't know if it's by design or not, but if a popular screen reader can't pick it up on the official ARIA page, there's nothing I can do.

Implementation is according to all specs and there's no action point for me here, afaik.

Can this issue be moved forward?

FYI @rak-phillip

**Comment by rak-phillip** [comment]
Is there any issue with the spec as it is implemented?

> Ensure that the sort state is programmatically defined for the sortable columns.
> Apply the following changes:
> • Provide the sort state using the "aria-sort" attribute on the element.
> o Set the value of "aria-sort" to "ascending" when the content is sorted in ascending order.
> o Set the value to "descending" when the content is sorted in descending order.
> o If no sorting is applied on the content, set the value to "none".
> o Ensure that the sort state is updated on user interaction.

Another review of #13448 shows that the tables implement the spec. Anything in addition to this to accommodate individual screen readers can be a new issue at this point; we'll need another round of triage to identify why one screen reader might identify the aria attributes while another may not.

**Comment by izaac** [comment]
@rak-phillip actually none of the screen readers has been picking the sorting text, As you can see on @aalves08 video he's forcing the OS X voice over to actually read the `HTML` code from the developer tools but not on the rendered element.

**Comment by izaac** [comment]
I don't think the sortable table sorting can be picked by the voice over tools after checking the `W3` documentation. Our UI implementation is following the exact expected behavior.  And the `aria-sort` is more related to the sorting logic of the table only. 

If there aren't any further concerns I'll continue validating what's remaining in the test execution.

**Comment by izaac** [comment]
Rancher version: `v2.11-0b081e077e830fa17b86151d21c786fb8d07c6de-head`
UI version: `master 0f09de2`
Firefox version: `135.0.1`
Chrome version: `133.0.6943.141`

I've validated the areas listed in the description, all the expected elements picked by the voice over (screen reader) were picked up as described in the `W3` `ARIA` documentation examples.
